scripts/lib_oval.py

```diff
@@ -502,6 +497,7 @@ def addElement(self, element, replace=True):
                 root.append(parent)
                 
             parent.append(element.getElement())
+            self.id_to_definition[ovalid] = element
```

Comment:
If you move the caches into getElementByID as suggested above, put this line under an `if hasattr(self, 'ovalid_to_element'):`

Reply:
And, you will only need one of these lines.